src/modules/content-s3/s3.c

```diff
@@ -0,0 +1,259 @@
+/************************************************************\
+ * Copyright 2020 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include <sys/types.h>
+#include <string.h>
+#include <errno.h>
+#include <stdio.h>
+#include <libs3.h>
+#include <stdlib.h>
+
+#include "src/common/libutil/errno_safe.h"
+
+#include "s3.h"
+
+static S3Status status_g;
```

Comment:
Suggestion: instead of a global, move this to `get_cb_data` and `put_cb_data`.  That will prepare the way later on for having more than one operation in flight, where we would allocate on struct per request.